libs/books/feature/src/lib/book-search/book-search.component.spec.ts

```diff
@@ -1,27 +1,56 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { async, ComponentFixture, discardPeriodicTasks, fakeAsync, TestBed, tick } from '@angular/core/testing';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { SharedTestingModule } from '@tmo/shared/testing';
 
 import { BooksFeatureModule } from '../books-feature.module';
 import { BookSearchComponent } from './book-search.component';
+import { provideMockStore, MockStore } from '@ngrx/store/testing';
 
 describe('ProductsListComponent', () => {
   let component: BookSearchComponent;
   let fixture: ComponentFixture<BookSearchComponent>;
+  let store: MockStore;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule]
+      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule],
+      providers: [
+        provideMockStore({
+          initialState: {
+            books: {
+              entities: []
+            }
+          }
+        })
+      ]
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(BookSearchComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    store = TestBed.inject(MockStore)
+    spyOn(component.searchForm, 'valueChanges');
+    spyOn(store, 'dispatch');
   });
 
   it('should create', () => {
     expect(component).toBeDefined();
   });
+
+  it('searchBook() should be called once the value changes and user waits for 500ms', fakeAsync(() => {
```

Comment:
Change it to

it('It should call searchBooks() when we have value for search term and actual time spent after entering search term is equal to 500ms', fakeAsync(() => {


change for other test as well in similar manner